src/components/Assets/SpecsComponent.jsx

```diff
@@ -0,0 +1,208 @@
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { Form, Grid } from 'semantic-ui-react';
+import range from 'lodash/range';
+import DropdownComponent from '../common/DropdownComponent';
+import ArtButton from '../common/ButtonComponent';
+
+import '../../_css/AddAssetComponent.css';
+
+class SpecsComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      specs: props.specs
+    };
+  }
+
+  generateYearRange = () => {
+    const currentYear = new Date().getFullYear();
+    const firstYearToConsider = 2011;
+
+    return range(currentYear, firstYearToConsider);
+  };
+
+  generateYearOptions = () => this.generateYearRange().map(year => ({
+    key: year,
+    text: year,
+    value: year
+  }));
+
+  onSelectYearOfManufacture = (event, data) => {
+    const specs = { ...this.state.specs };
+    specs.year = data.value;
+    this.setState({ specs });
+    this.props.pickRadioValuesFromSpecsComponent(specs);
+  };
+
+  handleRadioChanges = (event, data) => {
+    const { name, value } = data;
+    const specs = { ...this.state.specs };
+    if (name === 'processorType') {
+      specs.processorType = value;
+    } else if (name === 'processorSpeed') {
+      specs.processorSpeed = value;
+    } else if (name === 'screenSize') {
+      specs.screenSize = value;
+    } else if (name === 'storage') {
+      specs.storage = value;
+    } else if (name === 'memory') {
```

Comment:
The if statements are unnecessary.  Please read about [handling multiple inputs](https://reactjs.org/docs/forms.html#handling-multiple-inputs). The same logic can be applied in this instance. Ensure that the name of the radio buttons is the same with the state variables and have the logic as:
```jsx
this.setState(({ previousSpecs }) => (
  specs: {
    ...previousSpecs,
    [name]: <ValueOfRadioButton>
  }
))
```
Can you pass by my desk we discuss about `pickRadioValuesFromSpecsComponent` and how we can still make it work after the changes?